tools/ci-build/acquire-build-image

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+#
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0.
+#
+
+set -ue
+
+REMOTE_IMAGE_NAME="public.ecr.aws/w0m4q9l7/github-awslabs-smithy-rs-ci:latest"
```

Comment:
I'm a _little_ hesitant about always using latest as the tag of the image. If we fail to publish the docker image on main, we would never find out.

If we have main always tag with the commit hash, it seems like we could use the commit has of main here instead?